cob_generic_can/common/src/SocketCan.cpp

```diff
@@ -30,174 +51,147 @@
 bool SocketCan::receiveMsg(CanMsg* pCMsg)
 {
-    if (m_bInitialized == false) return false;
+    if (!m_bInitialized)
+    {
+        return false;
+    }
 
     bool bRet = false;
+    can::Frame frame;
 
-    if (recived) {
-        pCMsg->setID(recived_frame.id);
-        pCMsg->setLength(recived_frame.dlc);
-        pCMsg->set(recived_frame.data[0], recived_frame.data[1], recived_frame.data[2], recived_frame.data[3],
-                        recived_frame.data[4], recived_frame.data[5], recived_frame.data[6], recived_frame.data[7]);
+    if (m_reader.read(&frame, boost::chrono::seconds(1)))
+    {
+        pCMsg->setID(frame.id);
+        pCMsg->setLength(frame.dlc);
+        pCMsg->set(frame.data[0], frame.data[1], frame.data[2], frame.data[3],
+                   frame.data[4], frame.data[5], frame.data[6], frame.data[7]);
         bRet = true;
-	}
-	else {
-	  std::cout << "No message recived: " << std::endl;
-	}
-	return bRet;
+    }
+    return bRet;
 }
 
 //-------------------------------------------
 bool SocketCan::receiveMsgRetry(CanMsg* pCMsg, int iNrOfRetry)
 {
-    if (m_bInitialized == false) return false;
+    if (!m_bInitialized)
+    {
+        return false;
+    }
 
-    // wait until msg in buffer
+    can::Frame frame;
     bool bRet = false;
-    int i=0;
+    int i = 0;
+
     do
     {
-        if (recived) {
-            pCMsg->setID(recived_frame.id);
-            pCMsg->setLength(recived_frame.dlc);
-            pCMsg->set(recived_frame.data[0], recived_frame.data[1], recived_frame.data[2], recived_frame.data[3],
-                            recived_frame.data[4], recived_frame.data[5], recived_frame.data[6], recived_frame.data[7]);
+        if (m_reader.read(&frame, boost::chrono::microseconds(10)))
+        { 
+            pCMsg->setID(frame.id);
+            pCMsg->setLength(frame.dlc);
+            pCMsg->set(frame.data[0], frame.data[1], frame.data[2], frame.data[3],
+                       frame.data[4], frame.data[5], frame.data[6], frame.data[7]);
             bRet = true;
             break;
         }
         i++;
-        usleep(10000);
     }
-    while(i < iNrOfRetry);
+    while ((i < iNrOfRetry && bRet != true));
     return bRet;
 }
 
 //-------------------------------------------
 bool SocketCan::receiveMsgTimeout(CanMsg* pCMsg, int nSecTimeout)
 {
-    if (m_bInitialized == false) return false;
+    if (!m_bInitialized)
+    {
+        return false;
+    }
 
-    // wait until msg in buffer
     bool bRet = false;
+    can::Frame frame;
 
-    usleep(nSecTimeout/1000);
-    if (recived) {
-        pCMsg->setID(recived_frame.id);
-        pCMsg->setLength(recived_frame.dlc);
-        pCMsg->set(recived_frame.data[0], recived_frame.data[1], recived_frame.data[2], recived_frame.data[3],
-                        recived_frame.data[4], recived_frame.data[5], recived_frame.data[6], recived_frame.data[7]);
+    if (m_reader.read(&frame, boost::chrono::nanoseconds(nSecTimeout)))
```

Comment:
Please check against the other implementation, if nanoseconds are meant.
It would be nice, if you would add a comment to the base class to specify the correct unit.